substrate/frame/revive/src/evm.rs

```diff
@@ -37,33 +37,31 @@ pub fn extract_revert_message(exec_data: &[u8]) -> Option<String> {
 			let panic_code: u32 = U256::from_big_endian(exec_data.get(4..36)?).try_into().ok()?;
 
 			// See https://docs.soliditylang.org/en/latest/control-structures.html#panic-via-assert-and-error-via-require
-			let msg = match panic_code {
-				0x00 => "generic panic",
-				0x01 => "assert(false)",
-				0x11 => "arithmetic underflow or overflow",
-				0x12 => "division or modulo by zero",
-				0x21 => "enum overflow",
-				0x22 => "invalid encoded storage byte array accessed",
-				0x31 => "out-of-bounds array access; popping on an empty array",
-				0x32 => "out-of-bounds access of an array or bytesN",
-				0x41 => "out of memory",
-				0x51 => "uninitialized function",
-				code => return Some(format!("execution reverted: unknown panic code: {code:#x}")),
-			};
-
-			Some(format!("execution reverted: {msg}"))
+			return Some(match panic_code {
+				0x00 => "generic panic".to_string(),
+				0x01 => "assert(false)".to_string(),
+				0x11 => "arithmetic underflow or overflow".to_string(),
+				0x12 => "division or modulo by zero".to_string(),
+				0x21 => "enum overflow".to_string(),
+				0x22 => "invalid encoded storage byte array accessed".to_string(),
+				0x31 => "out-of-bounds array access; popping on an empty array".to_string(),
+				0x32 => "out-of-bounds access of an array or bytesN".to_string(),
+				0x41 => "out of memory".to_string(),
+				0x51 => "uninitialized function".to_string(),
+				code => format!("unknown panic code: {code:#x}"),
+			})
```

Comment:
No hard opinion on this, but do we want to maintain those on our own? There is the [alloy-sol-types](https://crates.io/crates/alloy-sol-types) crate which [defines them](https://github.com/alloy-rs/core/blob/e3071c893ac0ae7b80911764026371ea9544a3a3/crates/sol-types/src/types/error.rs#L310) and grepping through `polkadot-sdk` this crate is already in the dependency tree anyways.

Reply:
good point will look into this